src/components/RecordTxDialogUser/PrepareTxDialogUser.tsx

```diff
@@ -18,94 +16,75 @@ import {
   DialogTrigger,
 } from '@/components/ui/dialog.tsx';
 import { Input } from '@/components/ui/input.tsx';
-import { Label } from '@/components/ui/label.tsx';
 
 import { useSpvWalletClient } from '@/contexts';
 import { errorWrapper } from '@/utils';
+import { useForm } from 'react-hook-form';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Form, FormControl, FormDescription, FormField, FormItem, FormMessage, FormLabel } from '@/components/ui/form';
 
 interface RecordTxDialogProps {
   className?: string;
 }
 
 export const PrepareTxDialogUser = ({ className }: RecordTxDialogProps) => {
-  const [recipient, setRecipient] = useState<string>('');
   const [isPrepareDialogOpen, setIsPrepareDialogOpen] = useState(false);
-  const [recipientsBadges, setRecipientsBadges] = useState<string[]>([]);
-  const [amount, setAmount] = useState<string>('');
-  const [metadata, setMetadata] = useState(JSON.stringify({}));
 
   const { spvWalletClient } = useSpvWalletClient();
 
   const handleDialogToggle = () => {
     setIsPrepareDialogOpen((prev) => !prev);
+    form.reset();
   };
 
-  const handleMetadataChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
-    setMetadata(e.target.value);
-  };
-
-  const handleRecipientChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    setRecipient(e.target.value);
-  };
-
-  const handleAmountChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+  const formSchema = z.object({
+    recipient: z.string().min(3).email(),
+    amount: z.coerce.number().int().gte(1),
+    metadata: z
+      .string()
+      // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+      .transform((str, ctx): z.infer<ReturnType<any>> => {
+        try {
+          if (!str) {
+            return null;
+          }
+
+          return JSON.parse(str);
+        } catch (e) {
+          ctx.addIssue({ code: 'custom', message: 'Invalid JSON' });
+          return z.NEVER;
+        }
+      })
+      .nullable(),
+  });
+
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      recipient: '',
+      amount: 0,
+      metadata: '',
+    },
+  });
+
+  const onSubmit = async ({ recipient, amount, metadata }: z.infer<typeof formSchema>) => {
     try {
-      const convertedNumber = Number(e.target.value);
-      if (convertedNumber) {
-        setAmount(convertedNumber.toString());
-      }
-    } catch (err) {
-      console.error(err);
-      toast.warning('Amount should be a number');
-    }
-  };
+      const newRecipient: Recipient = { to: recipient, satoshis: amount };
 
-  const handleSubmit = async () => {
-    try {
-      const parsedMetadata = JSON.parse(metadata) as Metadata;
-      const recipients: Recipient[] = [];
+      await spvWalletClient?.DraftToRecipients([newRecipient], metadata);
```

Comment:
this would only prepare a draft of the transaction, which is not real transaction.
You should use SendToRecipients method.

Reply:
It was there before my changes. I assumed that someone had to review it earlier 😉